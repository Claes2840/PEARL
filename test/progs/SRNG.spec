// Internal vars
bit = 'nil
seedrev = 'nil
seedb = 'nil
tmp = 'nil

// Partial input, bitvector of size >= 2
seed = '(1 . (0 . (0 . (1 . nil))))

// Fully specialize
//count = '16
//nums = 'nil