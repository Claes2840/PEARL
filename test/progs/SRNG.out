init_9:
	entry
	if (count = '0)
		goto stop_9
		else loop_9

loop_9:
	fi nums
		from loop_4
		else init_9
	tmp ^= '(1.(0.(0.(1.nil))))
	nums <- (tmp . nums)
	count -= '1
	if (count = '0)
		goto stop_3
		else loop_3

loop_3:
	from loop_9
	assert(nums)
	tmp ^= '(0.(0.(1.(1.nil))))
	nums <- (tmp . nums)
	count -= '1
	if (count = '0)
		goto stop_6
		else loop_6

loop_6:
	from loop_3
	assert(nums)
	tmp ^= '(0.(1.(1.(0.nil))))
	nums <- (tmp . nums)
	count -= '1
	if (count = '0)
		goto stop_13
		else loop_13

loop_13:
	from loop_6
	assert(nums)
	tmp ^= '(1.(1.(0.(1.nil))))
	nums <- (tmp . nums)
	count -= '1
	if (count = '0)
		goto stop_10
		else loop_10

loop_10:
	from loop_13
	assert(nums)
	tmp ^= '(1.(0.(1.(0.nil))))
	nums <- (tmp . nums)
	count -= '1
	if (count = '0)
		goto stop_5
		else loop_5

loop_5:
	from loop_10
	assert(nums)
	tmp ^= '(0.(1.(0.(1.nil))))
	nums <- (tmp . nums)
	count -= '1
	if (count = '0)
		goto stop_11
		else loop_11

loop_11:
	from loop_5
	assert(nums)
	tmp ^= '(1.(0.(1.(1.nil))))
	nums <- (tmp . nums)
	count -= '1
	if (count = '0)
		goto stop_7
		else loop_7

loop_7:
	from loop_11
	assert(nums)
	tmp ^= '(0.(1.(1.(1.nil))))
	nums <- (tmp . nums)
	count -= '1
	if (count = '0)
		goto stop_15
		else loop_15

loop_15:
	from loop_7
	assert(nums)
	tmp ^= '(1.(1.(1.(1.nil))))
	nums <- (tmp . nums)
	count -= '1
	if (count = '0)
		goto stop_14
		else loop_14

loop_14:
	from loop_15
	assert(nums)
	tmp ^= '(1.(1.(1.(0.nil))))
	nums <- (tmp . nums)
	count -= '1
	if (count = '0)
		goto stop_12
		else loop_12

loop_12:
	from loop_14
	assert(nums)
	tmp ^= '(1.(1.(0.(0.nil))))
	nums <- (tmp . nums)
	count -= '1
	if (count = '0)
		goto stop_8
		else loop_8

loop_8:
	from loop_12
	assert(nums)
	tmp ^= '(1.(0.(0.(0.nil))))
	nums <- (tmp . nums)
	count -= '1
	if (count = '0)
		goto stop_1
		else loop_1

loop_1:
	from loop_8
	assert(nums)
	tmp ^= '(0.(0.(0.(1.nil))))
	nums <- (tmp . nums)
	count -= '1
	if (count = '0)
		goto stop_2
		else loop_2

loop_2:
	from loop_1
	assert(nums)
	tmp ^= '(0.(0.(1.(0.nil))))
	nums <- (tmp . nums)
	count -= '1
	if (count = '0)
		goto stop_4
		else loop_4

loop_4:
	from loop_2
	assert(nums)
	tmp ^= '(0.(1.(0.(0.nil))))
	nums <- (tmp . nums)
	count -= '1
	if (count = '0)
		goto stop_9
		else loop_9

exit_merge_0_6:
	fi (exit_control_var < '3)
		from exit_merge_0_2
		else exit_merge_3_6
	goto exit_merge_0_14

exit_merge_7_14:
	fi (exit_control_var < '11)
		from exit_merge_7_10
		else exit_merge_11_14
	goto exit_merge_0_14

exit_merge_0_2:
	fi (exit_control_var < '1)
		from stop_9
		else exit_merge_1_2
	goto exit_merge_0_6

exit_merge_3_6:
	fi (exit_control_var < '5)
		from exit_merge_3_4
		else exit_merge_5_6
	goto exit_merge_0_6

stop_9:
	fi nums
		from loop_4
		else init_9
	exit_control_var ^= '0
	goto exit_merge_0_2

exit_merge_1_2:
	fi (exit_control_var < '2)
		from stop_3
		else stop_6
	goto exit_merge_0_2

stop_3:
	from loop_9
	assert(nums)
	exit_control_var ^= '1
	goto exit_merge_1_2

stop_6:
	from loop_3
	assert(nums)
	exit_control_var ^= '2
	goto exit_merge_1_2

exit_merge_3_4:
	fi (exit_control_var < '4)
		from stop_13
		else stop_10
	goto exit_merge_3_6

exit_merge_5_6:
	fi (exit_control_var < '6)
		from stop_5
		else stop_11
	goto exit_merge_3_6

stop_13:
	from loop_6
	assert(nums)
	exit_control_var ^= '3
	goto exit_merge_3_4

stop_10:
	from loop_13
	assert(nums)
	exit_control_var ^= '4
	goto exit_merge_3_4

stop_5:
	from loop_10
	assert(nums)
	exit_control_var ^= '5
	goto exit_merge_5_6

stop_11:
	from loop_5
	assert(nums)
	exit_control_var ^= '6
	goto exit_merge_5_6

exit_merge_7_10:
	fi (exit_control_var < '9)
		from exit_merge_7_8
		else exit_merge_9_10
	goto exit_merge_7_14

exit_merge_11_14:
	fi (exit_control_var < '13)
		from exit_merge_11_12
		else exit_merge_13_14
	goto exit_merge_7_14

exit_merge_7_8:
	fi (exit_control_var < '8)
		from stop_7
		else stop_15
	goto exit_merge_7_10

exit_merge_9_10:
	fi (exit_control_var < '10)
		from stop_14
		else stop_12
	goto exit_merge_7_10

stop_7:
	from loop_11
	assert(nums)
	exit_control_var ^= '7
	goto exit_merge_7_8

stop_15:
	from loop_7
	assert(nums)
	exit_control_var ^= '8
	goto exit_merge_7_8

stop_14:
	from loop_15
	assert(nums)
	exit_control_var ^= '9
	goto exit_merge_9_10

stop_12:
	from loop_14
	assert(nums)
	exit_control_var ^= '10
	goto exit_merge_9_10

exit_merge_11_12:
	fi (exit_control_var < '12)
		from stop_8
		else stop_1
	goto exit_merge_11_14

exit_merge_13_14:
	fi (exit_control_var < '14)
		from stop_2
		else stop_4
	goto exit_merge_11_14

stop_8:
	from loop_12
	assert(nums)
	exit_control_var ^= '11
	goto exit_merge_11_12

stop_1:
	from loop_8
	assert(nums)
	exit_control_var ^= '12
	goto exit_merge_11_12

stop_2:
	from loop_1
	assert(nums)
	exit_control_var ^= '13
	goto exit_merge_13_14

stop_4:
	from loop_2
	assert(nums)
	exit_control_var ^= '14
	goto exit_merge_13_14

exit_merge_0_14:
	fi (exit_control_var < '7)
		from exit_merge_0_6
		else exit_merge_7_14
	exit
